src/go/rpk/pkg/serde/jsonschema_test.go

```diff
@@ -0,0 +1,444 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package serde
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/schemaregistry"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/sr"
+)
+
+const allTypesSchema = `{
+  "type": "object",
+  "properties": {
+    "name": { "type": "string" },
+    "id": { "type": ["string", "number"] },
+    "telephone": { "type": "integer" },
+    "member": { "type": "boolean" },
+    "apartments": { "type": "array" },
+    "tags": { "type": "object" },
+    "friends": { "type": "null" }
+  },
+  "required": ["name"]
+}`
+
+func Test_encodeDecodeJsonSchemaNoReferences(t *testing.T) {
+	tests := []struct {
+		name      string
+		schema    string
+		schemaID  int
+		record    string
+		expErr    bool // error building the Serde.
+		expEncErr bool // error encoding the record.
+	}{
+		{
+			name:     "Valid schema, record with all fields",
+			schema:   allTypesSchema,
+			schemaID: 2206,
+			record: `{
+  "name":"foo",
+  "id": 20.20,
+  "telephone": 12322212,
+  "member": true,
+  "apartments": [906, 2206, "test"],
+  "tags": { "random": "value" }
+}`,
```

Comment:
it might be worth putting these heredocs into consts at the top of the file for readability.

Reply:
I moved the biggest one to the top, but I think for the rest is worth keeping the records next to the schemas for table-driven tests. When we make a change, it is often better to have these closer to debug what's going on, and hopefully, once they are written we don't have to come back often to these files.